File: demo2/frontend/demo2/src/api/datasource.ts
/**
 * 数据源管理 API — 对应 ReportController 数据源端点
 */
import { apiAxios } from './src/requests'

export interface DatasourceEntity {
  id?: number
  name: string
  dbType: string
  host: string
  port: number
  dbName: string
  username: string
  description?: string
  password?: string
  params?: string
  status?: number
  createTime?: string
  updateTime?: string
}

export const ADatasource = {
  /** 数据源列表 — GET /report/datasource/list */
  list: apiAxios<DatasourceEntity[]>('/report/datasource/list', 'get'),

  /** 新增/编辑 — POST /report/datasource/save */
  save: apiAxios<number>('/report/datasource/save', 'post'),

  /** 删除 — DELETE /report/datasource/remove/{id} */
  remove: (id: number) =>
    apiAxios<string>(`/report/datasource/remove/${id}`, 'delete')(),

  /** 测试连接 — POST /report/datasource/test */
  test: apiAxios<boolean>('/report/datasource/test', 'post'),
}

File: demo2/src/main/java/com/example/domain/entity/Datasource.java
package com.example.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 数据源配置 — 对应 datasource 表
 */
@Data
@TableName("datasource")
public class Datasource {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 数据源名称 */
    private String name;

    /** 数据库类型：MYSQL / POSTGRESQL */
    private String dbType;

    /** 主机地址 */
    private String host;

    /** 端口 */
    private Integer port;

    /** 数据库名 */
    private String dbName;

    /** 用户名 */
    private String username;

    /** 密码（后续应做 AES 加密） */
    private String password;

    /** 额外 JDBC 连接参数 */
    private String params;

    /** 状态：1=启用，0=停用 */
    private Integer status;

    /** 软删除 */
    @TableLogic
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}
